File: mobile/ios/Sources/RoyalPMSMobile/Models/MaintenanceTicket.swift
import Foundation

struct MaintenanceTicket: Identifiable, Codable {
    let id: String
    let roomId: String?
    let roomNumber: String?
    let title: String
    let description: String?
    let priority: String
    let status: String
    let assignedTo: String?
    let reportedBy: String?
    let dueAt: String?
    let startedAt: String?
    let resolvedAt: String?
    let resolutionNotes: String?
    let createdAt: String
    let updatedAt: String?

    enum CodingKeys: String, CodingKey {
        case id, title, description, priority, status
        case roomId         = "room_id"
        case roomNumber     = "room_number"
        case assignedTo     = "assigned_to"
        case reportedBy     = "reported_by"
        case dueAt          = "due_at"
        case startedAt      = "started_at"
        case resolvedAt     = "resolved_at"
        case resolutionNotes = "resolution_notes"
        case createdAt      = "created_at"
        case updatedAt      = "updated_at"
    }

    var priorityLabel: String {
        switch priority {
        case "low":    return "Baixa"
        case "medium": return "Media"
        case "high":   return "Alta"
        case "urgent": return "Urgente"
        default:       return priority
        }
    }

    var statusLabel: String {
        switch status {
        case "open":        return "Aberto"
        case "in_progress": return "Em andamento"
        case "resolved":    return "Resolvido"
        case "cancelled":   return "Cancelado"
        default:            return status
        }
    }

    static let priorityOrder = ["urgent", "high", "medium", "low"]
}

File: mobile/android/app/src/main/java/com/royalpms/mobile/data/models/MaintenanceTicket.kt
package com.royalpms.mobile.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MaintenanceTicket(
    val id: String,
    @SerialName("room_id") val roomId: String? = null,
    @SerialName("room_number") val roomNumber: String? = null,
    val title: String,
    val description: String? = null,
    val priority: String,
    val status: String,
    @SerialName("assigned_to") val assignedTo: String? = null,
    @SerialName("reported_by") val reportedBy: String? = null,
    @SerialName("due_at") val dueAt: String? = null,
    @SerialName("started_at") val startedAt: String? = null,
    @SerialName("resolved_at") val resolvedAt: String? = null,
    @SerialName("resolution_notes") val resolutionNotes: String? = null,
    @SerialName("created_at") val createdAt: String,
    @SerialName("updated_at") val updatedAt: String? = null,
)

val MaintenanceTicket.priorityLabel get() = when (priority) {
    "low"    -> "Baixa"
    "medium" -> "Media"
    "high"   -> "Alta"
    "urgent" -> "Urgente"
    else     -> priority
}

val MaintenanceTicket.statusLabel get() = when (status) {
    "open"        -> "Aberto"
    "in_progress" -> "Em andamento"
    "resolved"    -> "Resolvido"
    "cancelled"   -> "Cancelado"
    else          -> status
}
